guides/common/modules/proc_configuring-project-with-keycloak-for-totp-authentication.adoc

```diff
@@ -0,0 +1,154 @@
+. In the *Bind Credential* field, enter your account password.
+. In the *Custom User LDAP Filter* field, enter _**_.
+. From the *Search Scope* list, select *Subtree*.
+. Click the *Test Connection* and *Test Authentication* button to ensure that configuration works.
+. Click *Save*.
+
+.Configuring Mappers in the {Keycloak} Web UI
+
+. Click the *Mappers* tab.
+. Click the *Create* button and create the following 7 mappers:
++
+.Mappers
+|===
+|Name |Mapper Type |User Model Attribute |LDAP Attribute |Read Only |Always Read Value From LDAP |Is Mandatory In LDAP |Is Binary Attribute |Password Policy Hints Enabled
+|username |user-attribute-ldap-mapper |username |sAMAccountName |ON |ON |ON |OFF |-
+|MSAD-account controls |mcad-user-account-control-mapper |- |- |- |- |- |- |ON
+|last name |user-attribute-ldap-mapper |lastName |sn |ON |ON |ON |OFF |-
+|first name |user-attribute-ldap-mapper |firstName |givenName |ON |ON |ON |OFF |-
+|modify date |user-attribute-ldap-mapper |modifyTimestamp |whenChanged |ON |ON |OFF |OFF |-
+|email |user-attribute-ldap-mapper |email |mail |ON |ON |OFF |OFF |-
+|creation date |user-attribute-ldap-mapper |creationTimestamp |whenCreated |ON |ON |OFF |OFF |-
+|===
+
+.Importing Users in the {Keycloak} Web UI
+
+. Navigate to *User Federation* and select the federation that you created.
+. Click the *Settings* tab.
+. Click *Synchronize All Users*.
+Synchronization might take some time on a production system when there is no ldap filter.
+When synchronization finishes, a green pop-up informing you that the process has completed is displayed.
+. In the realm menu at the right of the screen, navigate to *Users*.
+. Click *View all users* to view all imported users.
+
+.Configuring {Project}
+. On the {Project} server, install the following packages:
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# {package-install-project} mod_auth_openidc keycloak-httpd-client-install
+----
+
+. Register {Project} to {Keycloak} as a client:
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# keycloak-httpd-client-install --app-name foreman-openidc \
+--keycloak-server-url "https://{Keycloak-short}.com:8443" \
+--keycloak-admin-username "_admin_" \
+--keycloak-realm "_{Project}_Realm_" \
+--keycloak-admin-realm master \
+--keycloak-auth-role root-admin \
+-t openidc -l /users/extlogin --force
+----
++
+Enter the password for the administer account when prompted.
+. To configure {Project} to use {Keycloak} as an authentication source, enter the following command:
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# satellite-installer --foreman-keycloak true \
+--foreman-keycloak-app-name "foreman-openidc" \
+--foreman-keycloak-realm "_{Project}_Realm_"
+----
+
+.Configuring {Project] Client in the {Keycloak} Web UI
+. In the {Keycloak} web UI, navigate to {Clients} and click the {Project] client.
+. Ensure that the *Access Type* setting is set to *confidential*.
```

Comment:
The current RHSSO CAC cards docs say that the Acess Type must be set to public. What would work for both CAC and TOTP?

Reply:
Access Type is not related to the binding. We use `Access Type` as confidential for UI users and `Access Type` as public for CLI users.